### DISCUSSION TITLE
GNN with different MLP for second/third/etc neighbor

### DISCUSSION THREAD

**Discussion by antoni4040** [pyg-team/pytorch_geometric#3787] [Q&A]
Is there a GNN that provides different MLP layers for the first neighbors, for the second neighbors etc up to a certain degree? Seems to be a relatively simple concept that I struggle to find information about and I wonder if it could solve overfitting and oversmoothing for small graphs.

**Comment by rusty1s** [comment]
Do you mean something like https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.MFConv?

**Reply by antoni4040** [reply]
I'm not sure. What is the max_degree parameter? Is it the distance from each node? The paper seems to use different terms from the PyG documentation.

**Reply by rusty1s** [reply]
The `MFConv` will transform neighbors differently if they do not share the same node degree. Therefore, `max_degree` simply refers to the maximum node degree present in the input data. Let me know if this relates to your idea.
